spec/lib/va_profile/prefill/military_information_spec.rb

```diff
@@ -0,0 +1,364 @@
+            expect(subject.sw_asia_combat).to eq(false)
+          end
+        end
+      end
+
+      describe '#discharge_type' do
+        it 'returns discharge type' do
+          VCR.use_cassette('va_profile/military_personnel/post_read_service_history_200',
+                           match_requests_on: %i[method body]) do
+            expect(subject.discharge_type).to eq('general')
+          end
+        end
+
+        it 'with an unknown character_of_discharge_code it returns nil' do
+          # rubocop:disable RSpec/SubjectStub
+          allow(subject).to receive(:latest_service_episode).and_return(
+            OpenStruct.new(
+              character_of_discharge_code: nil
+            )
+          )
+          # rubocop:enable RSpec/SubjectStub
+
+          expect(subject.discharge_type).to eq(nil)
+        end
+      end
+
+      describe '#last_discharge_date' do
+        it 'returns last end date' do
+          VCR.use_cassette('va_profile/military_personnel/post_read_service_histories_200',
+                           match_requests_on: %i[method body]) do
+            expect(subject.last_discharge_date).to eq('2018-10-31')
+          end
+        end
+      end
+
+      describe '#last_entry_date' do
+        it 'returns last begin date' do
+          VCR.use_cassette('va_profile/military_personnel/post_read_service_histories_200',
+                           match_requests_on: %i[method body]) do
+            expect(subject.last_entry_date).to eq('2012-03-02')
+          end
+        end
+      end
+
+      describe '#post_nov111998_combat' do
+        context 'with no post 1998 deployment' do
+          it 'returns false' do
+            VCR.use_cassette('va_profile/military_personnel/post_read_service_histories_200',
+                             match_requests_on: %i[method body]) do
+              expect(subject.post_nov111998_combat).to eq(false)
+            end
+          end
+        end
+
+        context 'with a post 1998 deployment' do
+          it 'returns true' do
+            VCR.use_cassette('va_profile/military_personnel/post_read_service_history_200',
+                             match_requests_on: %i[method body]) do
+              expect(subject.post_nov111998_combat).to eq(true)
+            end
+          end
+        end
+      end
+
+      describe '#deployments' do
+        it 'returns deployments' do
+          VCR.use_cassette('va_profile/military_personnel/post_read_service_history_200',
+                           match_requests_on: %i[method body]) do
+            expect(
+              subject.deployments.pluck('deployment_end_date')
+            ).to eq(['2005-10-25'])
+          end
+        end
+      end
+
+      describe '#hca_last_service_branch' do
+        context 'with a nil service branch code' do
+          before do
+            # rubocop:disable RSpec/SubjectStub
+            expect(subject).to receive(:military_service_episodes).and_return(
```

Comment:
This is an expectation in a `before` block, which you don't want to have. Also, if you can find a way to generally write these specs without stubbing `subject`, that would be best. At best it sets you up for tests that are hooking too deep into implementation, and at worst you're just comparing stubs without actually comparing underlying code that you're trying to test

Reply:
We have removed the `before do` and turned this into an `allow`–seemed more appropriate. We noticed this in a test below, so we changed it there, too. 

We tried to avoid stubbing `subject` but we couldn't figure it out 😩. In the interest of time, we are going to keep this as-is. The good news is, in this instance, I'm pretty sure we're avoiding the worst-case scenario you laid out. An inadmissible `branch_of_service_code` from VA Profile is `'DVN'`, so we're making sure that returns `'other'`.